ksqldb-parser/src/main/java/io/confluent/ksql/schema/ksql/DefaultSqlValueCoercer.java

```diff
@@ -167,4 +208,162 @@ private static Result coerceMap(final Object value, final SqlMap targetType) {
 
     return Result.of(coerced);
   }
+
+  private interface ListObject {
```

Comment:
As previous comment, I don't think these classes are necessary (?)

Reply:
As above.

Merging this for now, will open a follow-up if it turns out I've missed something. Thanks!